pkg/github/github.go

```diff
@@ -108,6 +108,21 @@ func GetRepoAndOrgFromURL(repoURL string) (string, string, error) {
 	return repoName, orgName, nil
 }
 
+// GetDefaultBranchFromURL returns the default branch of a given repoURL
+func GetDefaultBranchFromRepo(repoURL string, client *github.Client, ctx context.Context) (string, error) {
+	repoName, orgName, err := GetRepoAndOrgFromURL(repoURL)
+	if err != nil {
+		return "", err
+	}
+	//repo.DefaultBranch
+	repo, _, err := client.Repositories.Get(ctx, orgName, repoName)
+	if err != nil {
+		return "", err
+	}
+
+	return *repo.DefaultBranch, nil
```

Comment:
I am not sure what values get returned for repo but repo is a pointer, is there a possibility of a panic if we dont check for nil?

Reply:
`repo` shouldn't be nil if an err isn't returned, but agreed, we should still check for nil

Reply:
updated.